contrib/kafka/src/main/java/com/google/cloud/dataflow/contrib/kafka/KafkaSource.java

```diff
@@ -0,0 +1,806 @@
+/*
```

Comment:
Could you rename this file `KafkaIO` and have the `PTransform<PBegin, [output type]>` be `KafkaIO.Read` or something similar? This follows examples like in `BigQueryIO`, `PubSubIO`, `BigtableIO`.

Hiding the source from the caller and exposing PTransforms generally makes the sources more usable and also allows for evolving the implementation over time.

(As an example, I'd like to rewrite `DatastoreIO.Read` based on `Create` instead of `BoundedSource`, but since we exposed the `BoundedSource` it's hard to do in a backwards-compatible way.)